### ISSUE TITLE
Ability to reload TLS certificates

### ISSUE THREAD

**Issue by jmallach** [42wim/matterircd#110]
Hi Wim,

I am using matterircd with TLS connections only, and the certificates are provided by Let's Encrypt, which means I need to refresh them every 2-3 months.

In order to avoid restarting matterircd everytime this happens, and kicking the users out, it would be nice if the server could reload new certificates upon receiving SIGHUP, as many other daemons do.

Bonus points if -tlsdir is replaced by -tlscert -tlskey, as Let's Encrypt has very specific certificate filenames: fullchain.pem, privkey.pem, etc. :)

**Comment by 42wim** [comment]
I'll take a look at it, maybe even integrate Let's Encrypt, go has some good libraries for this

**Comment by micah** [comment]
Hello,

I just came here to report this same bug. I am using Let's Encrypt just as @jmallach is, and I found that I needed to kick everyone out in order to load new certs. I also was hoping a simple HUP kill signal would reload the certs, but it just restarts everything.

I also was going to mention the same issue about the `-tlsdir` flag, I have to symlink my certs to the filenames that matterircd is looking for.

I realize that this was a couple years ago that this issue had activity, I'm wondering if any progress was made on it?
